File: solana/ts/src/matchingEngine/state/PreparedAuctionSettlement.ts
import { BN } from "@coral-xyz/anchor";
import { PublicKey } from "@solana/web3.js";

export class PreparedAuctionSettlement {
    bump: number;
    preparedBy: PublicKey;
    fastVaaHash: Array<number>;
    sourceChain: number;
    baseFee: BN;

    constructor(
        bump: number,
        preparedBy: PublicKey,
        fastVaaHash: Array<number>,
        sourceChain: number,
        baseFee: BN
    ) {
        this.bump = bump;
        this.preparedBy = preparedBy;
        this.fastVaaHash = fastVaaHash;
        this.sourceChain = sourceChain;
        this.baseFee = baseFee;
    }

    static address(
        programId: PublicKey,
        payer: PublicKey,
        fastVaaHash: Array<number> | Buffer | Uint8Array
    ) {
        return PublicKey.findProgramAddressSync(
            [Buffer.from("auction-settlement"), payer.toBuffer(), Buffer.from(fastVaaHash)],
            programId
        )[0];
    }
}

File: solana/programs/matching-engine/src/state/prepared_auction_settlement.rs
use anchor_lang::prelude::*;

#[account]
#[derive(Debug, InitSpace)]
pub struct PreparedAuctionSettlement {
    pub bump: u8,
    pub fast_vaa_hash: [u8; 32],

    pub prepared_by: Pubkey,

    pub source_chain: u16,
    pub base_fee: u64,
}

impl PreparedAuctionSettlement {
    pub const SEED_PREFIX: &'static [u8] = b"auction-settlement";
}
